tokio/src/sync/notify.rs

```diff
@@ -896,7 +896,7 @@ impl Drop for Notified<'_> {
         // This is where we ensure safety. The `Notified` value is being
         // dropped, which means we must ensure that the waiter entry is no
         // longer stored in the linked list.
-        if let Waiting = *state {
+        if matches!(*state, Waiting) {
```

Comment:
You've changed some of them to use `==` and some of them to use `matches!`. I would prefer to be consistent.